packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,98 @@
+import { ArnFormat, Lazy, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnInsightRule } from './cloudwatch.generated';
+
+/**
+ * Valid states for the rule.
+ */
+export enum RuleState {
+  /**
+   * Insight rule is on
+   */
+  ENABLED = 'ENABLED',
+
+  /**
+   * Insight rule is off
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * Propperties of defining a CloudWatch Contributor Insights rule.
+ */
+export interface InsightRuleProps {
+  /**
+   * The definition of the rule, as a JSON object.
+   */
+  readonly ruleBody: string;
```

Comment:
IMHO, this is not different than L1 construct or CFN written in JSON or YAML. We lose benefit of type check and abstraction as using `string` type here. Per [AWS doc](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContributorInsights-RuleSyntax.html), JSON key are fixed. In other words, only 5 keys, `Schema`, `LogGroupNames`, `LogFormat`, `Contribution`, and `AggregateOn`, are meaningful. Also, some keys allow specific values only. For instance, only `JSON` or `CLF` are valid values for `LogFormat`. How about introducing interface and making abstraction?

Reply:
I like what you're saying @jungseoklee. I agree.

Reply:
Probably an enum-like class, that also allows initialization with a string (and JSON object) is the best way to go.

How about:

```ts
RuleBody.fromString('{ ... }')
RuleBody.fromObject({
  ...
})
RuleBody.fromWhateverElse(...)
```